app/src/main/java/at/bitfire/icsdroid/db/AppDatabase.kt

```diff
@@ -43,49 +47,24 @@ import at.bitfire.icsdroid.db.entity.Subscription
 )
 abstract class AppDatabase : RoomDatabase() {
 
-    companion object {
-        @Volatile
-        private var instance: AppDatabase? = null
+    @Module
+    @InstallIn(SingletonComponent::class)
+    object AppDatabaseModule {
 
-        /**
-         * This function is only intended to be used by tests, use [getInstance], it initializes
-         * the instance automatically.
-         */
-        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-        fun setInstance(instance: AppDatabase?) {
-            this.instance = instance
-        }
-
-        /**
-         * Gets or instantiates the database singleton. Thread-safe.
-         * @param context The application's context, required to create the database.
-         */
-        fun getInstance(context: Context): AppDatabase {
-            // if we have an existing instance, return it
-            instance?.let {
-                return it
-            }
-
-            // multiple threads might access this code at once, so synchronize it
-            synchronized(AppDatabase) {
-                // another thread might just have created an instance
-                instance?.let {
-                    return it
+        @Provides
+        @Singleton
+        fun provideAppDatabase(
+            @ApplicationContext context: Context
+        ): AppDatabase = Room
+            .databaseBuilder(context, AppDatabase::class.java, "icsx5.db")
```

Comment:
You have changed the database name from `icsx5` to `icsx5.db`, so all data will be lost.